File: backend/src/main/java/io/github/jdiemke/ngforum/model/Post.java
package io.github.jdiemke.ngforum.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Entity
public class Post {

    @Id
    @GeneratedValue
    private Long id;

    private String name;
    private String content;

    public Post() {}

    public Post(String name, String content) {
        this.name = name;
        this.content = content;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

}

File: frontend/src/app/model/post.ts
import { autoserialize } from 'cerialize';

export class Post {

    @autoserialize
    public name: string;

    @autoserialize
    public content: string;

    public constructor(name: string, content: string) {
        this.name = name;
        this.content = content;
    }

    public getDisplayName(): string {
        return `${this.name} - ${this.content}`;
    }

}
